modules/pico_icmp6.c

```diff
@@ -58,39 +59,128 @@ uint16_t pico_icmp6_checksum(struct pico_frame *f)
 static void pico_icmp6_ping_recv_reply(struct pico_frame *f);
 #endif
 
+#ifdef PICO_SUPPORT_IPV6PMTU
+static void pico_icmp6_update_pmtu(struct pico_frame *f);
+#endif
+
 static int pico_icmp6_send_echoreply(struct pico_frame *echo)
 {
     struct pico_frame *reply = NULL;
     struct pico_icmp6_hdr *ehdr = NULL, *rhdr = NULL;
     struct pico_ip6 src;
     struct pico_ip6 dst;
+    static uint32_t frag_id = 0;
 
-    reply = pico_proto_ipv6.alloc(&pico_proto_ipv6, echo->dev, (uint16_t)(echo->transport_len));
-    if (!reply) {
-        pico_err = PICO_ERR_ENOMEM;
-        return -1;
+    {
+    uint32_t mtu = echo->dev->mtu;
+    const struct pico_ipv6_path_id path_id = {((struct pico_ipv6_hdr *)echo->net_hdr)->src};
+    const uint16_t header_overhead = (uint16_t) (echo->net_len + (echo->net_hdr - echo->buffer));
+
+#ifdef PICO_SUPPORT_IPV6PMTU
+    mtu = pico_ipv6_pmtu_get(&path_id);
+    if (mtu == 0) {
+        mtu = echo->dev->mtu;
+        pico_ipv6_path_add(&path_id, mtu);
+    }
+#endif
+
+    if (echo->transport_len < mtu - header_overhead) {
```

Comment:
I doubt TICS is going to like this function. since you have pretty giant if/else cases it might be worth putting some code in another function